src/shared/__tests__/libs/iota/transfers.spec.js

```diff
@@ -846,11 +847,20 @@ describe('libs: iota/transfers', () => {
         });
     });
 
-    // FIXME: The following test fails occasionally
     describe('#sortTransactionTrytesArray', () => {
         it('should sort transaction trytes in ascending order', () => {
             // failedTrytesWithCorrectTransactionHashes is in ascending order by default
-            const trytes = shuffle(failedTrytesWithCorrectTransactionHashes);
+            let shuffled = false;
+            let trytes = null;
+
+            // Ensures `trytes` never deeply equals `failedTrytesWithCorrectTransactionHashes`
+            // so that this test does not randomly fail
+            while (!shuffled) {
```

Comment:
I think we should try to avoid conditionals & loops in unit tests because they become hard to read and maintain. Each condition represent a test scenario of it's own. 

For this case specifically, I think using a `shuffled` list as an input in the first place was a bad idea. IMO, this should be simplified. 

I suggest just having the input list in opposite order and then asserting if it was sorted correctly. We can just [reverse](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse) the input trytes.

Reply:
For example:

```
it('should sort transaction trytes in ascending order', () => {
            // failedTrytesWithCorrectTransactionHashes is in ascending order by default
            const trytes = failedTrytesWithCorrectTransactionHashes.slice().reverse(); -> Changed this.
            const result = sortTransactionTrytesArray(trytes, 'currentIndex', 'asc');

           assert(result);
});
```

Reply:
That’s a good point. I can fix this tomorrow when I get some time

Reply:
Fixed in a4824bd